src/cloudsc_c/CMakeLists.txt

```
@@ -37,12 +45,19 @@ if( HAVE_CLOUDSC_C )
             cloudsc/cloudsc_validate.c
             cloudsc/mycpu.h
             cloudsc/mycpu.c
-        PUBLIC_INCLUDES
+	PRIVATE_INCLUDES
+	    # $<${HAVE_HDF5}:${HDF5_C_INCLUDE_DIRS}> # works on LUMI, doesn't work on AC
+	    $<${HAVE_HDF5}:string (REPLACE ";" " " DEST "${HDF5_C_INCLUDE_DIRS}")> # necessar on AC, not on LUMI
+	PUBLIC_INCLUDES
             $<INSTALL_INTERFACE:include>
             $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/cloudsc>
         PUBLIC_LIBS
-            Serialbox::Serialbox_C
-            $<${HAVE_OMP}:OpenMP::OpenMP_C>
+	    $<${HAVE_HDF5}:hdf5>
```

You should use here the target exported via `FindHDF5`:
```suggestion
	    $<${HAVE_HDF5}:hdf5::hdf5>
```

This will automatically inject include paths, link flags & co, so you can drop all of the `PRIVATE_INCLUDES` part and `link_directories` fuff.